### ISSUE TITLE
Widen dependencies

### ISSUE THREAD

**Issue by chhabrakadabra** [feast-dev/feast#2906]
**Is your feature request related to a problem? Please describe.**

Feast often needs to be installed alongside other libraries. Currently Feast pins many of its dependencies to the minor version instead of using a broader range. This means that if other libraries also pin to narrow version ranges, dependencies just can't be resolved.

**Describe the solution you'd like**

Change `setup.py` to use wider ranges for various dependencies. I'm not sure how that changes your testing strategy. Do you have to do some amount of matrix testing for important dependencies?

**Describe alternatives you've considered**

Not sure.

**Additional context**

N/A

**Comment by felixwang9817** [comment]
@chhabrakadabra thanks for raising this! I think we should probably just loosen all `==` dependencies to allow for all minor versions up to the next major version unless there is some explicit reason not to, as we did in #2647. For example, we would change `pygments==2.12.0` to `pygments>=2.12.0,<3`.

**Comment by kevjumba** [comment]
@chhabrakadabra if you would like to open a pr to pin the dependencies that would b egreat!

**Comment by kevjumba** [comment]
@chhabrakadabra Hey Abhin, one question, I see that you're pinning a bunch of versions to allow a major version upgrade. Is this necessary? E.g currently bigquery version is 2 and it would be easier if we pin it below 3?

Do you guys have specific needs for the next major version

**Comment by adchia** [comment]
We had a quick discussion. Seems like the preference is to throw errors if we see timezone naive timestamps and only use the `TIMESTAMP`

We can make this nicer in the future too with data source validation during `feast apply`

**Comment by chhabrakadabra** [comment]
> @chhabrakadabra Hey Abhin, one question, I see that you're pinning a bunch of versions to allow a major version upgrade. Is this necessary? E.g currently bigquery version is 2 and it would be easier if we pin it below 3?
> 
> Do you guys have specific needs for the next major version

Yeah, I think this is a sore spot for integrating with other libraries. The core idea behind this ticket is that Feast is being installed alongside many other libraries by data scientists that want to fetch data from the feature store. Many of these libraries have moved on to bigquery version 3 and even set a lower bound of 3.0.0. So Feast not upgrading the bigquery client library leads to people having to use older versions of these libraries.
